lit_gpt/lora.py

```diff
@@ -373,25 +366,16 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             Output tensor of shape (batch_size, context_length, 3 * embedding_size)
         """
 
-        def T(w):
-            return w.T if self.fan_in_fan_out else w
-
         # Let's assume that:
         # ⚬ x: (64, 64, 128) or (batch_size, context_length, embedding_size)
         # ⚬ self.weight: (384, 128) or (3 * embedding_size, embedding_size)
         # ⚬ self.lora_A.data: (4, 128)
         # ⚬ self.lora_B.data: (256, 2)
 
-        # the logic here is that the weights are merged only during inference
-        # so if they are merged we don't need to do anything with LoRA's A and B matrices
-        # but if the weights are not merged that means that the forward method is called during
-        # training and we need to forward pass input through pretrained weights, LoRA A and B matrices
-        # and do the summation (as per scheme at the top of the file)
-        if self.merged:
-            return F.linear(x, T(self.weight), bias=self.bias)
-        # `F.linear` automatically transposes the second argument (T(self.weight) in our case)
-        result = F.linear(x, T(self.weight), bias=self.bias)  # (64, 64, 128) @ (384, 128) -> (64, 64, 384)
-        if self.r > 0 and any(self.enable_lora):
+        # if weights are merged or LoRA is disabled (r <= 0 or all `enable_lora` are False) - it's only a regular nn.Linear forward pass;
+        # otherwise in addition do the forward pass with LoRA weights and add it's output to the output from pretrained weights
+        result = F.linear(x, self.T(self.weight), bias=self.bias)
+        if all((self.r > 0, any(self.enable_lora), not self.merged)):
```

Comment:
```suggestion
        if self.r > 0 and any(self.enable_lora) and not self.merged):
```
An argument can be made that this is more readable to the human

Reply:
With `all(...)` you can read it as:
> if `all` conditions are met then ...

With `and`:
> if this is true `and` this is true `and` ...

So I guess it's a personal preference. For me `all(...)` looks cleaner (and I usually prefer it when I need more than one `and` operator). 
But it's your repo, you are the boss, so ... 😄

Reply:
I'll apply the suggestion to keep consistency with the rest of the code base ([example](https://github.com/Lightning-AI/lit-gpt/blob/d0f082f164a8be5d3696b9e7d326b9d190ab9ac1/lit_gpt/model.py#L260C7-L265))

Reply:
Makes sense.